File: app-kotlin/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/views/investigation/sanity/SanityWarningLayout.java
package com.TritiumGaming.phasmophobiaevidencepicker.views.investigation.sanity;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;

import com.TritiumGaming.phasmophobiaevidencepicker.R;

public class SanityWarningLayout extends ConstraintLayout {

    public SanityWarningLayout(@NonNull Context context) {
        super(context);

        init(context, null);
    }

    public SanityWarningLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);

        init(context, attrs);
    }

    public SanityWarningLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        init(context, attrs);
    }

    public SanityWarningLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);

        init(context, attrs);
    }

    public void init(@NonNull Context c, @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs) {
        inflate(c, R.layout.layout_sanity_warnings, this);

        setDefaults();

        /*
        if (attrs != null) {
            TypedArray a = c.obtainStyledAttributes(attrs, R.styleable.PETImageButton);

            setImageResource(
                    a.getResourceId(
                            R.styleable.PETImageButton_PETImageButtonBackground,
                            R.drawable.icon_button_designs));

            setImageLevel(
                    a.getInt(
                            R.styleable.PETImageButton_PETImageButtonType, 0));

            a.recycle();
        }
        */

    }

    private void setDefaults() {
    }

}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/views/investigation/sanity/sanitywarn/SanityWarningLayout.kt
package com.TritiumGaming.phasmophobiaevidencepicker.views.investigation.sanity.sanitywarn

import android.content.Context
import android.util.AttributeSet
import androidx.constraintlayout.widget.ConstraintLayout
import com.TritiumGaming.phasmophobiaevidencepicker.R
import com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.EvidenceViewModel
import com.TritiumGaming.phasmophobiaevidencepicker.views.investigation.sanity.sanitywarn.alerts.SanityWarnActionView
import com.TritiumGaming.phasmophobiaevidencepicker.views.investigation.sanity.sanitywarn.alerts.SanityWarnHuntView
import com.TritiumGaming.phasmophobiaevidencepicker.views.investigation.sanity.sanitywarn.alerts.SanityWarnSetupView

class SanityWarningLayout : ConstraintLayout {

    lateinit var sanityWarnSetupView: SanityWarnSetupView
    lateinit var sanityWarnActionView: SanityWarnActionView
    lateinit var sanityWarnHuntView: SanityWarnHuntView

    constructor(context: Context) :
            super(context) { initView(null) }

    constructor(context: Context, attrs: AttributeSet?) :
            super(context, attrs) { initView(attrs) }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
            super(context, attrs, defStyleAttr) { initView(attrs) }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
            super(context, attrs, defStyleAttr, defStyleRes) { initView(attrs) }

    private fun initView(attrs: AttributeSet?) {
        inflate(context, R.layout.layout_sanity_warnings, this)

        sanityWarnSetupView = findViewById(R.id.evidence_sanitymeter_phase_setup)
        sanityWarnActionView = findViewById(R.id.evidence_sanitymeter_phase_action)
        sanityWarnHuntView = findViewById(R.id.evidence_sanitymeter_huntwarning)

        setDefaults()
    }

    private fun setDefaults() {
    }

    fun init(evidenceViewModel: EvidenceViewModel) {
        sanityWarnSetupView.init(evidenceViewModel)
        sanityWarnActionView.init(evidenceViewModel)
        sanityWarnHuntView.init(evidenceViewModel)
    }

}
